src/main/java/com/commercetools/sync/commons/utils/ClientConfigurationUtils.java

```diff
@@ -18,21 +19,31 @@
 import javax.annotation.Nonnull;
 import java.time.Duration;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.Random;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
 import static io.sphere.sdk.http.HttpStatusCode.BAD_GATEWAY_502;
 import static io.sphere.sdk.http.HttpStatusCode.GATEWAY_TIMEOUT_504;
 import static io.sphere.sdk.http.HttpStatusCode.SERVICE_UNAVAILABLE_503;
 
 public final class ClientConfigurationUtils {
-    private static HttpClient httpClient;
-    private static final long DEFAULT_TIMEOUT = 30000;
-    private static final TimeUnit DEFAULT_TIMEOUT_TIME_UNIT = TimeUnit.MILLISECONDS;
-    private static Map<SphereClientConfig, SphereClient> delegatesCache = new HashMap<>();
+
+    protected static final long DEFAULT_TIMEOUT = 30000;
+    protected static final long INITIAL_RETRY_DELAY = 200;
+    protected static final TimeUnit DEFAULT_TIMEOUT_TIME_UNIT = TimeUnit.MILLISECONDS;
+    protected static final int MAX_RETRIES = 5;
```

Comment:
Why only 5 times? We have in node-sdk 50 times: https://github.com/sphereio/sphere-node-sdk/blob/master/src/coffee/repeater-task-queue.coffee#L50

Reply:
50 🤔 this does not look right to me, with exponential backoff it will not be easy. 

Here the delays for the 10 retries.
  ````
            Retry 1: 318 millisecond
            Retry 2: 740 millisecond
            Retry 3: 1284 millisecond
            Retry 4: 2002 millisecond
            Retry 5: 6054 millisecond
            Retry 6: 8690 millisecond
            Retry 7: 15567 millisecond
            Retry 8: 30000 millisecond
            Retry 9: 30000 millisecond
            Retry 10: 30000 millisecond
````

Reply:
OK it's quite slow :| What if you keep 10 retries and don't do `final double exponentialFactor = Math.pow(2, retryAttempt - 1);` and do `INITIAL_RETRY_DELAY * retryAttempt * jitter`?

Reply:
```
         Retry 8: 30000 millisecond
          Retry 9: 30000 millisecond
          Retry 10: 30000 millisecond
```
This does not seem right and might cause snowball effect. It should be always randomly growing. Maybe just compare if generated delay is bigger or equal default timeout and then stop retrying instead?

Reply:
Not sure how I could stop the retry, 
In our defaults, it's 5, so those cases are not possible. when we plan a retry with a configurable max retry count like 10 in this case, that might be an issue, in this case, I will increase the maximum timeout or might not put a max constant for the delay. Let me check.

Reply:
We have our upper bound set as `30000`. Once we reach this limit, we will no longer increase our delay.